### ISSUE TITLE
Failed gbasf2 grid download because failed_files.txt contains directory

### ISSUE THREAD

**Issue by meliache** [nils-braun/b2luigi#112]
My task was all successful and tried to download the job outputs, but it failed with the following message:
```
Downloading remaining files from dataset with command  gb2_ds_get --force /belle/user/meliache/2021-06-106ac9f82805/sub00/upsilon_4s_*.root --input_dslist /nfs/dust/belle2/user/meliache/git_hash=light-2002-ichep/
job_name=2021-06-10/input_files=hashed_eb2e36a81e191f2c9fe4b2962bb9de29/mc=True/fei_charge_type=B0/upsilon_4s.root.partial/failed_files.txt
No file found

/afs/desy.de/user/m/meliache/.local/lib/python3.6/site-packages/b2luigi/batch/processes/gbasf2.py:325: RuntimeWarning: RuntimeError('No output data for gbasf2 project 2021-06-106ac9f82805 found.',)
  warnings.warn(repr(err), RuntimeWarning)
```
I checked the output of the `failed_files.txt` and it only contains the line
```
/belle/user/meliache/2021-06-106ac9f82805/sub00/upsilon_4s_00039_job194473040_00.root in /nfs/dust/belle2/user/meliache/git_hash=light-2002-ichep/job_name=2021-06-10/input_files=hashed_eb2e36a81e191f2c9fe4b2962bb9de29/mc=True/fei_charge_type=B0/upsilon_4s.root.partial/2021-06-106ac9f82805/sub00
```
Seems to be caused by PR #91 by @ArturAkh, which introduced this log file and included some command output parsing, which might have failed. As I assume from looking at the helper method `_failed_files_from_dataset_download` it should only contain root files, but maybe something went wrong in all the string splitting?

**Comment by ArturAkh** [comment]
Hi @meliache,

Yes, it may well be that this is because of parsing, since this seems to be a wrong line taken into the text file.

Do you use the newest `gbafs2` release by now? (which may have changed the printout format from v5.0.2 to v5.1.0)

In case you still have the printout in your screen/tmux session at the point where the first download attempt was started (so the one leading to creation of the file `failed_files.txt`), it would be good if you could send it to me.

For the time-being, I'd suggest to remove the file & restart the download again (reproducing previous behaviour of trying to download all output files again).

**Comment by meliache** [comment]
My `gbasf2KEK/defaults-Belle-KEK.cfg` file tells me `Release = v5r0p2`, to which I added some patch from an email exchange in which you were involved, because my jobs were stalled. I searched comp-user-forum and found you having the same problem with the FEI and the issue was that gbasf2 fails for too large standard outputs if I remember correctly and some patch was provided. Anyway, I moved the old `.partial` directory to another file name and it proceeded the download correctly.

I'm aware that having the actual output is important for debugging, I just was in hurry because I had some Sunday plans and just wanted to check on my grid jobs and so I quickly fixed this for myself, basically created the issue to not forget that and then went outside. If I find the outputs I'll provide them.

I think it would be good to add a check that the file in the list of failed files all end with `.root` (I think gbasf2 doesn't support anything else anyway) and if not, issue a warning and not create that file.

**Comment by ArturAkh** [comment]
Ok, sounds good!

I'll implement such a consistency check of the content to be written to the file & make a pull request in the next days.

**Comment by meliache** [comment]
@ArturAkh Any updates? @bilokin also reported to me issues when downloading, I asked him to check the `failed_files.txt`and according to him it had the contents
```
/belle/user/bilokin/L0_b16bv1ee107e827f/sub00/L0_b16bv1_00096_job196745307_00.root in /gpfs/group/belle2/dataprod/Systematics/systematic_corrections_framework/syscorrfw/git_hash=release-05-02-06/dst_type=mdst/mode
l_name=Lambda0/forMC=False/gbasf2_project_name_prefix=L0_b16bv1/gbasf2_input_dataset=hashed_f8320b26d30ab433c5a54546d21f414c/gbasf2_input_dslist=hashed_2770681b48bafb9f4fae24d9592f7ba2/L0_b16bv1.root.partial/L0_b16bv1ee107e827f/sub00
```
bilokins original question passed to me in the BelleII Chat was:

> Hi, do you know how to deal with the incomplete downloads of gbasf2 projects in b2luigi? For example, now I have a project, where one file was not downloaded from the Grid, for some reason and the gbasf2 task was failed because of that. After a while I can manually download a file, but if I simply restart b2luigi it still fails instantly. The first fail logs are:
> ```
> Failed files:
> /belle/user/bilokin/L0_b16bv1ee107e827f/sub00/L0_b16bv1_00096_job196745307_00.root in /gpfs/group/belle2/dataprod/Systematics/systematic_corrections_framework/syscorrfw/git_hash=release-05-02-06/dst_type=mdst/model_name=Lambda0/forMC=False/gbasf2_project_name_prefix=L0_b16bv1/gbasf2_input_dataset=hashed_f8320b26d30ab433c5a54546d21f414c/gbasf2_input_dslist=hashed_2770681b48bafb9f4fae24d9592f7ba2/L0_b16bv1.root.partial/L0_b16bv1ee107e827f/sub00
> 
> 
> 
> /home/belle2/bilokin/.local/lib/python3.6/site-packages/b2luigi/batch/processes/gbasf2.py:615: RuntimeWarning: 
> Files missing in download:
> L0_b16bv1_00096_job196745307_00.root
>   warnings.warn("\nFiles missing in download:\n{}".format("\n".join(missing_files)), RuntimeWarning)
> /home/belle2/bilokin/.local/lib/python3.6/site-packages/b2luigi/batch/processes/gbasf2.py:322: RuntimeWarning: RuntimeError('Download incomplete. The downloaded set of files in /gpfs/group/belle2/dataprod/Systematics/systematic_corrections_framework/syscorrfw/./git_hash=release-05-02-06/dst_type=mdst/model_name=Lambda0/forMC=False/gbasf2_project_name_prefix=L0_b16bv1/gbasf2_input_dataset=hashed_f8320b26d30ab433c5a54546d21f414c/gbasf2_input_dslist=hashed_2770681b48bafb9f4fae24d9592f7ba2/L0_b16bv1.root.partial/L0_b16bv1ee107e827f/sub00 is not equal to the list of dataset files on the grid for project L0_b16bv1ee107e827f.',)
>   warnings.warn(repr(err), RuntimeWarning)
> 
> Task SysAnalysisTask failed!
> Parameters
> 	 git_hash = release-05-02-06
> 	 num_processes = 0
> 	 max_event = 0
> 	 dst_type = mdst
> 	 model_name = Lambda0
> 	 forMC = False
> 	 gbasf2_project_name_prefix = L0_b16bv1
> 	 gbasf2_input_dataset = False
> 	 gbasf2_input_dslist = lpns/bucket16b_all_lpns.list
> Please have a look into the log files in
> /gpfs/group/belle2/dataprod/Systematics/systematic_corrections_framework/syscorrfw/logs/git_hash=release-05-02-06/dst_type=mdst/model_name=Lambda0/forMC=False/gbasf2_project_name_prefix=L0_b16bv1/gbasf2_input_dataset=hashed_f8320b26d30ab433c5a54546d21f414c/gbasf2_input_dslist=hashed_2770681b48bafb9f4fae24d9592f7ba2/SysAnalysisTask
> 
> INFO: Informed scheduler that task   SysAnalysisTask_mdst_False_False_8b21b0256b   has status   FAILED
> ```
I'm tempted to make the "improved" dataset download optional via a setting and maybe disable it by default, since the string parsing results in errors and additional code complexity, is hard to maintain accross releases, compared to just letting gbasf2 handling the downloads (which will hopefully improve in the future). One can override the `failed_files.txt` file database, which fixes many download bugs, but this is not clear to users.

**Comment by ArturAkh** [comment]
Hi @meliache, sorry, I was busy with other tasks unfortunately and today I'm working towards freezing features for basf2 6 release.

So I only will be able to solve that earliest by tomorrow.

**Comment by ArturAkh** [comment]
@meliache, please have a look at #113 which should fix the problem

**Comment by meliache** [comment]
Closed by #113
